Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -582,6 +624,87 @@ private extension HTTPClient {
 
 }
 
+// MARK: - Request Retry Logic
+extension HTTPClient {
+    internal func retryRequestIfNeeded(
```

Comment:
This is the only function I wasn't able to directly test - let me know if y'all have any ideas! 😄